dbms/src/AggregateFunctions/AggregateFunctionGroupUniqArray.cpp

```diff
@@ -1,13 +1,36 @@
 #include <DB/AggregateFunctions/AggregateFunctionFactory.h>
 #include <DB/AggregateFunctions/AggregateFunctionGroupUniqArray.h>
 #include <DB/AggregateFunctions/Helpers.h>
+#include <common/logger_useful.h>
 
 namespace DB
 {
 
 namespace
 {
 
+static IAggregateFunction * createWithExtraTypes(const IDataType & argument_type)
+{
+		 if (typeid_cast<const DataTypeDateTime *>(&argument_type))	return new AggregateFunctionGroupUniqArray<UInt32>;
+	else if (typeid_cast<const DataTypeDate *>(&argument_type)) 	return new AggregateFunctionGroupUniqArray<UInt16>;
+	else
```

Comment:
More clear is to use `DataTypeDateTime::FieldType` and `DataTypeDate::FieldType`.